### PR TITLE
Use typescript + React hooks

### PR DESCRIPTION
**Pull request by bruiztorres**
Main goal of this PR is improving typing and using new React API. I also upgraded some dependencies. 

This is the first step to provide some functionalities that are available in `powerbi-client` but not exposed in this library.

### CODE REVIEW

**Comment by howlowck** [comment]
Hi @bruiztorres Thanks for the contribution. I will take a look this week.

File: src/index.tsx [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+'accessToken' |
+'permissions'> & {
+  onEmbedded?: (embed: Embed) => void;
+  embedType: EmbedType;
+  mobile?: boolean;
+  filterPaneEnabled?: boolean;
+  navContentPaneEnabled?: boolean;
+};
+
+export default function PowerbiEmbedded ({ onEmbedded, ...props }: PowerbiEmbeddedProps): JSX.Element {
```
Comment (howlowck) [inline src/index.tsx:30]: I'm not a fan of namespaces. Can you change `JSX.Element` to `React.FunctionComponentElement<PowerbiEmbeddedProps>`
OR better yet, change the name function to an arrow function like so:
`const PowerbiEmbedded: React.FC<PowerbiEmbeddedProps> = ({ onEmbedded, ...props }) => ...`

Comment (bruiztorres) [reply src/index.tsx]: 👍

**Comment by howlowck** [comment]
I've merged your changes, and have since published it along with some other changes to 0.7.0

Thank you for your contribution!!

**Pull request merged**